SU2_CFD/src/fluid/CFluidFlamelet.cpp

```diff
@@ -0,0 +1,245 @@
+  }
+
+  source_scalar.resize(n_scalars);
+  lookup_scalar.resize(n_lookups);
+
+  Pressure = value_pressure_operating;
+
+  PreprocessLookUp();
+}
+
+CFluidFlamelet::~CFluidFlamelet() {
+  delete look_up_table;
+}
+
+/*--- do a lookup for the list of variables in table_lookup_names, for visualization purposes ---*/
+unsigned long CFluidFlamelet::SetScalarLookups(const su2double* val_scalars) {
+  su2double enth = val_scalars[I_ENTH];
+  su2double prog = val_scalars[I_PROGVAR];
+
+  string name_enth = table_scalar_names[I_ENTH];
+  string name_prog = table_scalar_names[I_PROGVAR];
+
+  /*--- perform table lookup ---*/
+  unsigned long exit_code = look_up_table->LookUp_XY(table_lookup_names, lookup_scalar, prog, enth);
+
+  return exit_code;
+}
+
+/*--- set the source terms for the transport equations ---*/
+unsigned long CFluidFlamelet::SetScalarSources(const su2double* val_scalars) {
+  table_sources[0] = 0.0;
+
+  string name_enth = table_scalar_names[I_ENTH];
+  string name_prog = table_scalar_names[I_PROGVAR];
+
+  /*--- value for the progress variable and enthalpy ---*/
+  su2double enth = val_scalars[I_ENTH];
+  su2double prog = val_scalars[I_PROGVAR];
+
+  /*--- perform table lookup ---*/
+  unsigned long exit_code = look_up_table->LookUp_XY(varnames_Sources, val_vars_Sources, prog, enth);
+
+  /*--- the source term for the progress variable is always positive by construction, but we clip from below  --- */
+  source_scalar[I_PROGVAR] = max(EPS, table_sources[I_SRC_TOT_PROGVAR]);
+  source_scalar[I_ENTH] = 0.0;
+
+  /*--- source term for the auxiliary species transport equations---*/
+  for (size_t i_aux = 0; i_aux < n_user_scalars; i_aux++) {
+    /*--- The source term for the auxiliary equations consists of a production term and a consumption term:
+          S_TOT = S_PROD + S_CONS * Y ---*/
+    su2double y_aux = val_scalars[n_control_vars + i_aux];
+    su2double source_prod = table_sources[1 + 2 * i_aux];
+    su2double source_cons = table_sources[1 + 2 * i_aux + 1];
+    source_scalar[n_control_vars + i_aux] = source_prod + source_cons * y_aux;
+  }
+
+  return exit_code;
+}
+
+void CFluidFlamelet::SetTDState_T(su2double val_temperature, const su2double* val_scalars) {
+  su2double val_enth = val_scalars[I_ENTH];
+  su2double val_prog = val_scalars[I_PROGVAR];
+
+  string name_enth = table_scalar_names[I_ENTH];
+  string name_prog = table_scalar_names[I_PROGVAR];
+
+  /*--- add all quantities and their address to the look up vectors ---*/
+  look_up_table->LookUp_XY(varnames_TD, val_vars_TD, val_prog, val_enth);
+
+  /*--- compute Cv from Cp and molar weight of the mixture (ideal gas) ---*/
+  Cv = Cp - UNIVERSAL_GAS_CONSTANT / molar_weight;
+}
+
+/* --- Total enthalpy is the transported variable, but we usually have temperature as a boundary condition,
+       so we do a reverse lookup */
+unsigned long CFluidFlamelet::GetEnthFromTemp(su2double* val_enth, const su2double val_prog, const su2double val_temp,
```

Comment:
Don't pass scalar variables by pointer please, and in general output arguments should come after the inputs.
```suggestion
unsigned long CFluidFlamelet::GetEnthFromTemp(su2double& val_enth, const su2double val_prog, const su2double val_temp,
```